shapeshift/web
# feat: thorchain savers withdraw

## Description

This PR handles THORChain savers withdraws - memoless for the time being.

To be handled as stack on top of this one: 

- The status screen doesn't yet show dust/fees. We will need some additional routing logic for that: given at this point the Tx is already ongoing and the quote is final, we should *not* do the same quote-fetching logic we're doing here

- Confirmed working:
  - [ ] BTC
  - [x] BCH
  - [x] ATOM
  - [ ] DOGE
  - [x] LTC
  - [ ] ETH
  - [x] AVAX

## Notice




## Pull Request Type


## Issue (if applicable)

N/A

## Risk



None, isolated to savers

## Testing



- Enable the savers flag
- Withdraw a percentage of your position that's not close to the max (can be user-input or a percent option)
- Wait for the Tx to be confirmed on-chain
- Reload the app, wait for the position to refresh, and ensure funds are deducted from your DeFi position and back in your wallet

### Engineering



☝🏽 

### Operations



☝🏽 

## Screenshots (if applicable)

### BCH

<img width="361" alt="image" src="https://user-images.githubusercontent.com/17035424/213745959-6fa0005d-6464-46b6-8919-ac5c91a024fc.png">
<img width="518" alt="image" src="https://user-images.githubusercontent.com/17035424/213745929-aa8a3d71-69df-468e-b85e-4547ecc00163.png">
<img width="522" alt="image" src="https://user-images.githubusercontent.com/17035424/213745768-2f584c6f-4855-4834-a859-2b679c345bab.png">
<img width="536" alt="image" src="https://user-images.githubusercontent.com/17035424/213745071-289250cf-2bbb-489c-842d-d51fc9a066d4.png">
<img width="351" alt="image" src="https://user-images.githubusercontent.com/17035424/213761966-3ab64bdf-6483-4ce5-a10b-3c697b74eaa3.png">


### LTC

<img width="356" alt="image" src="https://user-images.githubusercontent.com/17035424/213762498-c6215c04-6844-4484-b858-11b8e5e099ed.png">
<img width="534" alt="image" src="https://user-images.githubusercontent.com/17035424/213762541-e9cdf610-862b-41aa-b93a-2dbfdcdbcf6b.png">
<img width="521" alt="image" src="https://user-images.githubusercontent.com/17035424/213762595-9eb69fa8-afd3-4fe7-a2f1-9824a4e560e2.png">
<img width="536" alt="image" src="https://user-images.githubusercontent.com/17035424/213762679-64a11924-ff54-49be-bb48-524a887612a3.png">
<img width="367" alt="image" src="https://user-images.githubusercontent.com/17035424/213765821-091597aa-6e5c-438f-b590-8d9ae49539f4.png">


### ATOM

<img width="418" alt="image" src="https://user-images.githubusercontent.com/17035424/213773591-e6ee4b23-94b0-42e4-bb15-c1c33db6cf03.png">
<img width="508" alt="image" src="https://user-images.githubusercontent.com/17035424/213773631-1ba0acca-2f82-4be8-9e2b-5abe451b1c40.png">
<img width="522" alt="image" src="https://user-images.githubusercontent.com/17035424/213773664-caa05764-20e9-4358-bc76-bb166b15ca84.png">
<img width="538" alt="image" src="https://user-images.githubusercontent.com/17035424/213773743-920a2cee-cc7d-461d-871f-09d6c6b7906c.png">
<img width="430" alt="image" src="https://user-images.githubusercontent.com/17035424/213773961-9abd5651-47fb-48d9-9733-5a2d54366363.png">

### AVAX

<img width="364" alt="image" src="https://user-images.githubusercontent.com/17035424/213786707-acbfb982-175c-4077-a5c6-604530c805c0.png">
<img width="529" alt="image" src="https://user-images.githubusercontent.com/17035424/213786733-919f36c3-4599-4d4a-8137-9fee20bc6291.png">
<img width="532" alt="image" src="https://user-images.githubusercontent.com/17035424/213786759-c7e2f64c-783a-419f-8ea5-aa4a6ab345ba.png">
<img width="547" alt="image" src="https://user-images.githubusercontent.com/17035424/213786811-a5b18c9f-ee66-4f3a-8f57-4c7e93a090d6.png">
<img width="355" alt="image" src="https://user-images.githubusercontent.com/17035424/213786969-2e47d2c4-f44a-4ebb-b81d-313da60d185f.png">
